### PR TITLE
Add default return format to the context

### LINKED ISSUE #6281: Make possibility to setup return format in global web3 instance class.
The main problem for users to move from 1.x to 4.x is returning format. All numbers become a BigInt. So, it will be great to have the possibility to set a default returning format in the web3 instance and then all methods in all packages (eth, contract, etc) will return correct formatting

### PR DESCRIPTION
**Pull request by avkos**
## Description

Please include a summary of the changes and be sure to follow our [Contribution Guidelines](https://github.com/web3/web3.js/blob/4.x/.github/CONTRIBUTING.md).

Fixes #6281 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run lint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:coverage` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have linked Issue(s) with this PR in "Linked Issues" menu.

### CODE REVIEW

File: packages/web3-eth-contract/src/contract.ts [hunk 1]
```diff
@@ -799,7 +799,8 @@ export class Contract<Abi extends ContractAbi>
 			},
 			estimateGas: async <ReturnFormat extends DataFormat = typeof DEFAULT_RETURN_FORMAT>(
 				options?: PayableCallOptions,
-				returnFormat: ReturnFormat = DEFAULT_RETURN_FORMAT as ReturnFormat,
+				returnFormat: ReturnFormat = (this.defaultReturnFormat ??
+					DEFAULT_RETURN_FORMAT) as ReturnFormat,
```
Comment (jdevcs) [inline packages/web3-eth-contract/src/contract.ts]: need a test for format here

File: packages/web3-eth/src/utils/get_revert_reason.ts [hunk 2]
```diff
@@ -80,7 +85,8 @@ export async function getRevertReason<
 	web3Context: Web3Context<EthExecutionAPI>,
 	transaction: TransactionCall,
 	contractAbi?: ContractAbi,
-	returnFormat: ReturnFormat = DEFAULT_RETURN_FORMAT as ReturnFormat,
+	returnFormat: ReturnFormat = (web3Context.defaultReturnFormat ??
```
Comment (jdevcs) [inline packages/web3-eth/src/utils/get_revert_reason.ts]: as now we have default format coming from context so there isn't need of DEFAULT_RETURN_FORMAT form types package, and same of other places in this PR.

File: packages/web3-eth/src/web3_eth.ts [hunk 3]
```diff
@@ -219,7 +220,8 @@ export class Web3Eth extends Web3Context<Web3EthExecutionAPI, RegisteredSubscrip
 	 * ```
 	 */
 	public async getHashRate<ReturnFormat extends DataFormat = typeof DEFAULT_RETURN_FORMAT>(
-		returnFormat: ReturnFormat = DEFAULT_RETURN_FORMAT as ReturnFormat,
+		returnFormat: ReturnFormat = (this.defaultReturnFormat ??
+			DEFAULT_RETURN_FORMAT) as ReturnFormat,
```
Comment (jdevcs) [inline packages/web3-eth/src/web3_eth.ts]:
```suggestion
			web3Context.defaultReturnFormat) as ReturnFormat,
```
for all places as now we have update-able default set in context so that should be used instead of `DEFAULT_RETURN_FORMAT` from types package, so:
if some one is explicitly setting format via this.defaultReturnFormat it will work, but if not it will send DEFAULT_RETURN_FORMAT to rpc-wrapper, so rpc-wrapper will get value and will not use context default? in that case if its `local ?? context` then it will reflect format settings that can be updated.

**Review by jdevcs — APPROVED** [review]
(edited)
@avkos precedence will be following:
1. local function format ( if defined will be considered first )
2. else context format
for all packages

File: packages/web3-eth-ens/test/integration/ens.events.test.ts [hunk 4]
```diff
@@ -120,15 +119,6 @@ describeIf(isSocket)('ens events', () => {
 		else provider = new ENS.providers.HttpProvider(clientUrl);
 
 		ens = new ENS(registry.options.address, provider);
-
-		const block = await getBlock(ens, 'latest', false, DEFAULT_RETURN_FORMAT);
```
Comment (luu-alex) [inline packages/web3-eth-ens/test/integration/ens.events.test.ts:124]: why was this removed? edit: i see you don't set gas, instead just using type

File: docs/docs/guides/web3_config/index.md [hunk 5]
```diff
@@ -198,4 +199,46 @@ console.log(web3.getContextObject().config)
   transactionPollingTimeout: 750000,
   ...
 */
-```
\ No newline at end of file
+```
+
+
+### defaultReturnFormat
```
Comment (luu-alex) [inline docs/docs/guides/web3_config/index.md:205]: Think there should be a guide about defaultReturnFormat under upgrading from 1.x too because this PR is targetted towards users upgrading

**Review by luu-alex — APPROVED** [review]
